custom_components/spotcast/helpers.py

```diff
@@ -103,117 +107,324 @@ async def run(*args, loop=None, executor=None, **kwargs):
             )
+        )
+        == 0
+    ):
+        searchResults = get_top_tracks(artistName, spotify_client)
+        _LOGGER.debug("Playing top tracks for artist: %s", searchResults[0]["name"])
+        return searchResults
+    else:
+        searchString = get_search_string(
+            artistName=artistName,
+            albumName=albumName,
+            trackName=trackName,
+            genreName=genreName,
+            playlistName=playlistName,
+            showName=showName,
+            episodeName=episodeName,
+            audiobookName=audiobookName,
+        )
+
+        searchTypes = get_types_string(
+            artistName=artistName,
+            albumName=albumName,
+            trackName=trackName,
+            playlistName=playlistName,
+            showName=showName,
+            episodeName=episodeName,
+            audiobookName=audiobookName,
+        )
+        searchResults = spotify_client.search(
+            q=searchString, limit=limit, offset=0, type=searchTypes, market=country
+        )
+
+        compiledResults = []
+        if "tracks" in searchResults:
+            for item in searchResults["tracks"]["items"]:
+                compiledResults.append(item)
+        if "albums" in searchResults:
+            for item in searchResults["albums"]["items"]:
+                compiledResults.append(item)
+        if "playlists" in searchResults:
+            for item in searchResults["playlists"]["items"]:
+                compiledResults.append(item)
+        if "shows" in searchResults:
+            for item in searchResults["shows"]["items"]:
+                compiledResults.append(item)
+        if "audiobooks" in searchResults:
+            for item in searchResults["audiobooks"]["items"]:
+                compiledResults.append(item)
+        if "episodes" in searchResults:
+            for item in searchResults["episodes"]["items"]:
+                compiledResults.append(item)
+
+        _LOGGER.debug(
+            "Found %d results for %s. First Track name: %s",
+            len(compiledResults),
+            searchString,
+            compiledResults[0]["name"],
+        )
+
+        return compiledResults
+
+
+def search_tracks(
+    search: str,
+    spotify_client: spotipy.Spotify,
+    appendToQueue: bool = False,
+    shuffle: bool = False,
+    startRandom: bool = False,
+    limit: int = 20,
+    artistName: str = None,
+    country: str = None,
+):
+    results = get_search_results(search, spotify_client, artistName, limit, country)
+    if len(results) > 0:
+        firstResult = [results[0]]
+        if not startRandom:
+            results = results[1:limit]
+        if shuffle:
+            random.shuffle(results)
+        if not startRandom:
+            results = firstResult + results
+
+    return results
+
+
+def add_tracks_to_queue(
+    spotify_client: spotipy.Spotify, tracks: list = [], limit: int = 20
+):
+    filtered = list(filter(lambda x: x["type"] == "track", tracks))
```

Comment:
Notice above how we return other entities like albums along with tracks. Spotify won't let you enqueue and album, only a track, so we filter out tracks here (before applying the limit)